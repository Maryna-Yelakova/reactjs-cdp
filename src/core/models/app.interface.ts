export interface AppState {
      query:string;
      list:Array<MovieModel>;
      movie: MovieModel;
}

export interface MovieModel {
      id: number;
      title: string;
      tagline: string;
      vote_average: number;
      vote_count: number;
      release_date: string;
      poster_path: string;
      overview: string;
      budget: number;
      revenue: number;
      runtime: number;
      genres:Array<string>;
}
