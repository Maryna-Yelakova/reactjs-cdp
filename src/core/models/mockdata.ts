export const MockMovie = {
      id: 0,
      title: '',
      tagline: '',
      vote_average: 0,
      vote_count: 0,
      release_date: '',
      poster_path: '',
      overview: '',
      budget: 0,
      revenue: 0,
      runtime: 0,
      genres: ['']
}

export const MockMovieList = [
  {
    id: 1,
    title: 'Imperial concubine',
    tagline: 'comedy',
    vote_average: 2045,
    vote_count: 2045,
    release_date: '',
    poster_path: '',
    overview: '',
    budget: 0,
    revenue: 0,
    runtime: 0,
    genres: ['']
  },
  {
    id: 2,
    title: 'Meng Qi Shi Shen',
    tagline: 'romantic',
    vote_average: 764,
    vote_count: 764,
    release_date: '',
    poster_path: '',
    overview: '',
    budget: 0,
    revenue: 0,
    runtime: 0,
    genres: ['']
  }
]
