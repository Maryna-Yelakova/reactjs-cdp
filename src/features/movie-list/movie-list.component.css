.movie_list {
	display: flex;
  flex-wrap: wrap;
	justify-content: space-around;
	background-color: #efefef;
	color: #333;
  padding: 40px;
  box-sizing: border-box;
}

.movie_list_item {
  width: 25%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.movie_list_info {
  padding: 10px 0;
}

.movie_list_name {
  font-weight: bold;
  margin-bottom: 10px;
}
