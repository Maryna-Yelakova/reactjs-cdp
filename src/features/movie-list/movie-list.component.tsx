import * as React from 'react';
import './movie-list.component.css';
import { AppContext } from '../../context';

export const MovieList = ({selectMovie, list}) => {
  return (
    // <AppContext.Consumer>
    // {context =>
      <ul className="movie_list">
        {list.map((movie) => <li className="movie_list_item" key={movie.id} onClick={() => selectMovie(movie.id)}>
          <div className="movie_list_item-content">
            <img src={ movie.poster_path } width="200" height="302" alt="picture"/>
            <div className="movie_list_info">
              <p className="movie_list_name">{movie.title}</p>
              <i className="movie_list_genre">{movie.vote_count}</i>
            </div>
          </div>
        </li> )}
      </ul>
    // }
    // </AppContext.Consumer>
  );
}
// https://medium.com/@RubenOostinga/avoiding-deeply-nested-component-trees-973edb632991
