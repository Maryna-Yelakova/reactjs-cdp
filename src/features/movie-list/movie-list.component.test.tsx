import * as  React from 'react';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { MovieList } from './movie-list.component';
import { MockMovieList } from '../../core/models/mockdata';


describe('MovieList', () => {
 it('should render correctly', () => {
   const mockSelectMovie = jest.fn();
   const component = shallow(<MovieList list={MockMovieList} selectMovie={mockSelectMovie} />);

   expect(component).toBeDefined();
   expect(shallowToJson(component)).toMatchSnapshot();
 });

 it('handle click correctly', () => {
   const mockSelectMovie = jest.fn();

  const component = shallow(<MovieList list={MockMovieList} selectMovie={mockSelectMovie} />);
  const item = component.find('.movie_list_item').first();
  item.simulate('click');

  expect(mockSelectMovie).toHaveBeenCalled();
 })
});
