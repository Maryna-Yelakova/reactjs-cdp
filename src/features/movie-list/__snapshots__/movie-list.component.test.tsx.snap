// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should render correctly 1`] = `
<ul
  className="movie_list"
>
  <li
    className="movie_list_item"
    key="1"
    onClick={[Function]}
  >
    <div
      className="movie_list_item-content"
    >
      <img
        alt="picture"
        height="302"
        src={
          Object {
            "default": "test-file-stub",
          }
        }
        width="200"
      />
      <div
        className="movie_list_info"
      >
        <p
          className="movie_list_name"
        >
          Imperial concubine
        </p>
        <i
          className="movie_list_genre"
        >
          2045
        </i>
      </div>
    </div>
  </li>
  <li
    className="movie_list_item"
    key="2"
    onClick={[Function]}
  >
    <div
      className="movie_list_item-content"
    >
      <img
        alt="picture"
        height="302"
        src={
          Object {
            "default": "test-file-stub",
          }
        }
        width="200"
      />
      <div
        className="movie_list_info"
      >
        <p
          className="movie_list_name"
        >
          Meng Qi Shi Shen
        </p>
        <i
          className="movie_list_genre"
        >
          764
        </i>
      </div>
    </div>
  </li>
</ul>
`;
